src/components/DiffView/index.spec.tsx

```diff
@@ -276,12 +272,55 @@ describe(__filename, () => {
     });
   });
 
+  it('does not tokenize the hunks to add syntax highlighting if the diff has been trimmed', () => {
+    const _codeCanBeHighlighted = jest.fn().mockReturnValue(true);
+    const _codeShouldBeTrimmed = jest.fn().mockReturnValue(true);
+    const _tokenize = jest.fn();
+
+    renderWithLinterProvider({
+      _codeCanBeHighlighted,
+      _codeShouldBeTrimmed,
+      _tokenize,
+      // A minified file will be trimmed by default.
+      isMinified: true,
+    });
+
+    expect(_tokenize).not.toHaveBeenCalled();
+  });
+
+  it('resets tokens to undefined if they seem invalid', () => {
+    const _codeCanBeHighlighted = jest.fn().mockReturnValue(true);
+    const _codeShouldBeTrimmed = jest.fn().mockReturnValue(false);
+    const _tokenize = (jest.spyOn(
+      ReactDiffView,
+      'tokenize',
+    ) as unknown) as typeof ReactDiffView.tokenize;
+    const badCode =
```

Comment:
Is it really bad code or partial code that will break the tokenize function? Maybe we should have a comment here too. This comment could also say that if this test case starts to fail, one should look into the return value of the `tokenize` function.

Reply:
Good idea about the variable name change, and I also added a detailed comment above the assertion.